Frontend/frontend/src/Util/APIRequest.js

```diff
@@ -28,4 +28,36 @@ export default class APIRequest {
 
         return 'Bearer ' + User.getUserState().token;
     }
+
+    //Checks if the username already exists in the database
+    static async checkNewAccount(username, email) {
+        let res = await fetch(apiBaseUrl + "api/user/check",
+        {
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            method: "POST",
+            body: JSON.stringify({ username: username, email: email})
+        });
+
+        return res;
+    }
+
+    static async register(username, firstName, lastName, dateOfBirth, phoneNumber,
+        email, medicalInfo, bankAccountNumber, password) {
+        let res = await fetch(apiBaseUrl + "api/user/register",
+        {
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            method: "POST",
+            body: JSON.stringify({ username: username, firstname: firstName, lastname: lastName,
+                                    dateofbirth: dateOfBirth, phonenumber: phoneNumber, email: email,
+                                    medicalinformation: medicalInfo, bankAccountNumber: bankAccountNumber, password: password})
```

Comment:
Change `bankAccountNumber: bankAccountNumber` to `bankAccount: bankAccountNumber`
Right now when registering, the bank account number gets set to null in the database.  
This is because in the backend, the field name in the User model is named bankAccount. Both field names have to match.

From Bryan & Richard

Reply:
Fixed, thanks for catching that

Reply:
LGTM